src/client.rs

```diff
@@ -65,20 +66,33 @@ impl<T> HttpsConnector<T> {
     pub fn https_only(&mut self, enable: bool) {
         self.force_https = enable;
     }
-    
+
+    /// Set the host which should be used for Server Name Indication (SNI).
+    ///
+    /// if the `sni_host` is not set the default host from URI will be used
+    pub fn sni_host(&mut self, host: &str) {
```

Comment:
The SNI host is already sent normally, just extracted from the `Uri` hostname. So this would really be more an *override*, right?

Reply:
Another concern that can be had is that this would force *all* requests to always use the same SNI value. It seems simple to accidentally configure this option and then use the connector for several different domains.

Reply:
> The SNI host is already sent normally, just extracted from the `Uri` hostname. So this would really be more an _override_, right?

Yes, this is correct. In some cases we need  to override the hostname with different value. 

> Another concern that can be had is that this would force _all_ requests to always use the same SNI value. It seems simple to accidentally configure this option and then use the connector for several different domains.

yeah, this is the valid concern, but I haven't found a better way how I could provide the different hostname for the TLS connection.  Maybe I missed  something here, but if I could have access to the underlying `Request`  I could tell http connector to extract the domain from the `Host` header instead if provided (in this case it might be need to have such an option exposes to configure http connecter before building the client), which might be a bit better way to set this up.